include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -82,13 +83,13 @@ MultigridPreconditionerBase<dim, Number, MultigridNumber>::initialize(
 
   this->initialize_matrix_free_objects();
 
-  this->initialize_operators();
+  this->initialize_transfer_operators();
 
-  this->initialize_smoothers();
+  this->initialize_operators();
 
-  this->initialize_coarse_solver(operator_is_singular);
+  this->initialize_smoothers(initialize_preconditioners);
 
-  this->initialize_transfer_operators();
+  this->initialize_coarse_solver(operator_is_singular, initialize_preconditioners);
 
```

Comment:
Why is the sequence here changed as well? Does that have a reason?

Reply:
Not strictly necessary, but I wanted to clarify the code by first doing the initialization steps not depending on `initialize_preconditioners`, and then do the rest that depends on `initialize_preconditioners`.